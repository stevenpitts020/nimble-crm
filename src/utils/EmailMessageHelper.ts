/**
 * This file contains helpers for the accept/Decline modals
 *
 * Use it to DRY your code. Peace.
 */
const EmailMessageHelper = {
  buildDeclineEmailMessage: (messageFields: {
    template: string | undefined
    bankName: string | undefined
    fullName: string | undefined
    productName: string | undefined
    employeeName: string | undefined
    bankSlug: string | undefined
  }) => {
    let text = '' + messageFields.template
    text = text.replace(/\{bankName\}/g, messageFields.bankName || '')
    text = text.replace(/\{name\}/g, messageFields.fullName || '')
    text = text.replace(/\{product\}/g, messageFields.productName || '')
    text = text.replace(/\{employee\}/g, messageFields.employeeName || '')
    text = text.replace(/\{domain\}/g, messageFields.bankSlug || '')
    return text
  },
  buildAproveEmailMessage: (messageFields: {
    template: string | undefined
    routingNumber: string | undefined
    bankName: string | undefined
    amount: number | undefined
    accountNumber: string | undefined
    fullName: string | undefined
    productName: string | undefined
    employeeName: string | undefined
    bankSlug: string | undefined
  }) => {
    let text = '' + messageFields.template
    text = text.replace(/\{accountNumber\}/g, messageFields.accountNumber || '')
    text = text.replace(/\{routingNumber\}/g, messageFields.routingNumber || '')
    text = text.replace(/\{bankName\}/g, messageFields.bankName || '')
    text = text.replace(/\{amount\}/g, messageFields.amount?.toString(10) || '')
    text = text.replace(/\{name\}/g, messageFields.fullName || '')
    text = text.replace(/\{product\}/g, messageFields.productName || '')
    text = text.replace(/\{employee\}/g, messageFields.employeeName || '')
    text = text.replace(/\{domain\}/g, messageFields.bankSlug || '')
    return text
  },
}
export default EmailMessageHelper
