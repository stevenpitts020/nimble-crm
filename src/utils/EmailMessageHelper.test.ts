import EmailMessageHelper from './EmailMessageHelper'

describe('EmailMessageHelper', () => {
  describe('buildAproveEmailMessage', () => {
    const mockTemplate =
      '{accountNumber}, {amount}, {routingNumber}, {bankName}, {domain}, {name}, {product}, {employee}'

    test('should return text', () => {
      const props = {
        bankName: 'Central Bank',
        bankSlug: 'centralbank.com',
        employeeName: 'Mark',
        routingNumber: 'xxx',
        amount: 2000,
        accountNumber: '000',
        fullName: 'Steve',
        productName: 'Simple Checking',
        template: mockTemplate,
      }
      const result = EmailMessageHelper.buildAproveEmailMessage(props)

      expect(result).toMatchSnapshot()
    })
  })

  describe('buildDeclineEmailMessage', () => {
    const declineTemplate =
      '{bankName}, {domain}, {name}, {product}, {employee}'
    test('should return text', () => {
      const props = {
        bankName: 'Central Bank',
        employeeName: 'Mark',
        bankSlug: 'centralbank.com',
        accountNumber: '000',
        fullName: 'Steve',
        productName: 'Simple Checking',
        template: declineTemplate,
      }
      const result = EmailMessageHelper.buildDeclineEmailMessage(props)
      expect(result).toMatchSnapshot()
    })
  })
})
