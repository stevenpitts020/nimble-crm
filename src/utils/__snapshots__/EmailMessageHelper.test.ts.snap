// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailMessageHelper buildAproveEmailMessage should return text 1`] = `"000, 2000, xxx, Central Bank, centralbank.com, Steve, Simple Checking, Mark"`;

exports[`EmailMessageHelper buildDeclineEmailMessage should return text 1`] = `"Central Bank, centralbank.com, Steve, Simple Checking, Mark"`;
